### DISCUSSION TITLE
How to manage CRC Cluster configuration snapshots.

### DISCUSSION THREAD

**Discussion by eduizquierdo** [crc-org/crc#2907] [Q&A] [answered]
Hi, I found CRC really usefull for testing & development. But, because all the cluster and deployments are running in the local host, resources are limited and very soon you reach the limits.
It would be very useful having a way to mange snapshots, so you can save your current cluster configuration and start a fresh one where you can add more resources without hitting the limits. Apparently, doesn’t look a feature very complex to develop as all the cluster configuration in inside the virtual machine and CRC already provides commands to create a new virtual machine. The one thing missing is commands to save&restore current VM before creating a new one (`crc --save-snapshot `and `crc --restore-snapshot`).
Like this feature is not available (or I'm not aware of it), I’m looking for a way to do it manually. 
**Is there a safe way for creating a copy of the VM and restoring later as to be used by “CRC start”?**

**Comment by gbraad** [comment, marked as answer]
We do not handle snapshots as crc by itself is meant as an ephemeral setup.
You could however rely on `virsh` to make snapshots (on Linux), but that would be handled outside of CRC.

---

```
$ virsh snapshot-create-as --domain crc --name "before-deployment"
```

```
$ virsh snapshot-list --domain crc
```

For more see: https://libvirt.org/manpages/virsh.html#snapshot-commands

Note: this can have effect on the way the CRC command works, as when a snapshot exists the VM can not be deleted using the regular `crc delete` command. This is because the snapshots are in a different location from the actual VM. In that case you would need to delete each snapshot and associated metadata first:

```
$ sudo virsh snapshot-delete --metadata crc <snapshot-name>
```
